specs-go/config.go

```diff
@@ -79,6 +85,8 @@ type Platform struct {
 	OS string `json:"os"`
 	// Arch is the architecture
 	Arch string `json:"arch"`
+	// OSVersion is the version of the operating system. (this field is platform dependent)
+	OSVersion string `json:"os.version,omitempty" platform:"windows"`
```

Comment:
We use it today in docker/docker for difference in functionality between versions of Windows as opposed to whether things are supported or not supported. Hence it's not a case of us being able to return an error in the way described above.